functions/controllers/retrieveM2MToken/index.js

```diff
@@ -0,0 +1,30 @@
+const functions = require("firebase-functions");
+const fetch = require("node-fetch");
+const { db } = require("../../utils/admin");
+
+const getM2MToken = () => {
+  options = {
```

Comment:
You should take context as an argument and this should allow you to pass this directly into the pubsub scheduler.